### PR TITLE
Pull requests should be connected to filed issues

### LINKED ISSUE #1699: Pull Requests should be attached to an existing issue
We should dissuade contributors (including ourselves) from sending PRs that aren't related to an existing issue. An issue should _always_ be filed first that explains the problem and motivation. The PR should then reference the issue number.

### PR DESCRIPTION
**Pull request by benwerd**
This way we can ensure a PR has a single purpose. There's no obvious way to _require_ that a PR is attached to an existing issue - so we should include it in the template instead.

Fixes #1699

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [x] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by hardlydifficult** [comment]
Would it make sense to add this to the checklist?  Something like
- [ ] Relevant issue IDs are mentioned in the description.

**Comment by benwerd** [comment]
@hardlydifficult I added it as:

[ ] It references and cleanly addresses an existing, filed issue

But I'm inclined to add "ID"

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 <!--
-Please include a summary of the change and which issue is fixed -include its number-. Please also include relevant motivation and context.
+Please include a summary of the change and which issue is fixed -include its number-. It's important that PRs connect to an existing issue, and we'll review this PR in part based on the content of that issue. Please also include relevant motivation and context.
 -->
 
 # Checklist:
 
 - [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
+  - [ ] It references and cleanly addresses an existing, filed issue
```
Comment (cellog) [inline PULL_REQUEST_TEMPLATE.md]:
```suggestion
  - [ ] This PR fixes Issue #<!-- replace with relevant issue -->
```

Comment (hardlydifficult) [reply PULL_REQUEST_TEMPLATE.md]: I sort of like it, sort of don't.  Often a PR is related to, but does not completely address, an issue.  Git will auto close an issue if you say `fixes #x`... although I'm not sure that applies with the word issue in between.

Comment (cellog) [reply PULL_REQUEST_TEMPLATE.md]: yes, perhaps better is

```suggestion
- [ ] This PR addresses Issue #<!-- replace with relevant issue -->
```

**Comment by hardlydifficult** [comment]
> @hardlydifficult I added it as:

Sorry for some reason I didn't see that line the first time =p

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (benwerd) [reply PULL_REQUEST_TEMPLATE.md]: I gave IDs their own sections. What do you think?

Comment (hardlydifficult) [reply PULL_REQUEST_TEMPLATE.md]: I like it.  But now is the description change and checklist addition overkill?

Comment (benwerd) [reply PULL_REQUEST_TEMPLATE.md]: Fair! I removed the checklist item.

**Comment by benwerd** [comment]
Altered a little bit to add a distinct issue number section. PTAL?

**Pull request merged**